crates/ruff_workspace/src/options.rs

```diff
@@ -333,6 +333,19 @@ pub struct Options {
     )]
     pub target_version: Option<PythonVersion>,
 
+    /// A list of mappings from file pattern to Python version to use when checking the
+    /// corresponding file(s).
```

Comment:
We might want to extend this documentation by mentioning that it is to override the global default `target-version` or `requires-python` and mention a use case where this is desired (and what the implications are)